penguin/project/workflow_orchestrator.py

```
@@ -148,9 +144,34 @@ async def run_next_task(self, project_id: Optional[str] = None) -> Optional[Dict
             exec_result = await self.task_executor.execute_task(task)
             self._debug(f"Execution result: {exec_result}")
-            if exec_result["status"] == "error":
+            executor_status = exec_result.get("status")
+            if (
+                executor_status is None
+                or not isinstance(executor_status, str)
+                or executor_status.strip() == ""
+            ):
+                raise ValueError(
+                    f"Malformed executor result for task {task.id} ({task.title}): {exec_result!r}"
+                )
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)